### PR TITLE
[NPU] update inference cmake, test=develop

### PR DESCRIPTION
**Pull request by qili93**
### PR types

New features

### PR changes

Others

### Describe

Update inference_lib.cmake for NPU, and also add Ascend Toolkit and Driver version.

### CODE REVIEW

File: cmake/inference_lib.cmake [hunk 1]
```diff
@@ -364,6 +365,11 @@ function(version version_file)
                 "HIP version: ${HIP_VERSION}\n"
                 "MIOpen version: v${MIOPEN_MAJOR_VERSION}.${MIOPEN_MINOR_VERSION}\n")
     endif()
+    if(WITH_ASCEND_CL)
+        file(APPEND ${version_file}
+                "Ascend Toolkit version: ${ASCEND_TOOLKIT_VERSION}\n"
+                "Ascend Driver version: ${ASCEND_DRIVER_VERSION}\n")
+    endif()
```
Comment (jiweibo) [inline cmake/inference_lib.cmake:373]: 不打开 WITH_ASCEND_CXX11， C++预测实际上没法用，可以考虑把这个也加到version.txt中

**Review by jiweibo — DISMISSED** [review]
LGTM

File: cmake/inference_lib.cmake (hunk 1 above)
Comment (qili93) [reply cmake/inference_lib.cmake:373]: Done, added "WITH_ASCEND_CXX11: ${WITH_ASCEND_CXX11}\n"

**Review by jiweibo — DISMISSED** [review]
LGTM

**Review by zhiqiu — APPROVED** [review]
LGTM

**Pull request merged**